src/main/scala/org/broadinstitute/dsde/workbench/gpalloc/monitor/ProjectCreationSupervisor.scala

```diff
@@ -15,13 +15,28 @@ import org.broadinstitute.dsde.workbench.gpalloc.util.Throttler
 
 import scala.concurrent.Future
 import scala.concurrent.duration._
+import scala.util.Random
 
 object ProjectCreationSupervisor {
+  var gpAllocConfig: GPAllocConfig = null
+
+  private def randomProjectName(): String = {
+    //strip out things GCP doesn't like in project IDs: uppercase, underscores, and things that are too long
+    val sanitizedPrefix = gpAllocConfig.projectPrefix.toLowerCase.replaceAll("[^a-z0-9-]", "").take(22)
+    s"$sanitizedPrefix-${Random.alphanumeric.take(7).mkString.toLowerCase}"
+  }
+
   sealed trait ProjectCreationSupervisorMessage
   case class RequestNewProject(projectName: String) extends ProjectCreationSupervisorMessage
+  object RequestNewProject {
+    def apply(): RequestNewProject = {
+      RequestNewProject(randomProjectName())
```

Comment:
alternatively, make a `RandomNewProject` message and send that. i kinda prefer this version as it's a little clearer from calling code